src/engine/sync/synccontrol.cpp

```diff
@@ -484,11 +491,14 @@ void SyncControl::updateAudible() {
 }
 
 void SyncControl::slotRateChanged() {
-    double bpm = m_pLocalBpm ? m_pLocalBpm->get() * m_pRateRatio->get() : 0.0;
+    mixxx::Bpm bpm = getLocalBpm();
+    if (bpm.isValid()) {
```

Comment:
it looks like it would be simpler to early-return if bpm is invalid.

Reply:
Unsure if the rate ratio could be NaN/infinity. Is this a read-only CO? Otherwise Bpm may become invalid due to the multiplication. I restructured the code a bit.